obsidian-tasks-group/obsidian-tasks
# test: Add more sample data for testing Cache - and streamline its creation

# Description



New test data:

- Add more sample Markdown files to the `Tasks-Demo` vault, in  `Test Data/*.md`
- Add TypeScript versions of these files in `tests/Obsidian/__test_data__/*.ts`

Streamlined creation and processing of this test data:

- Add a Templater template that may help with creating a new file, for single-tasks cases:       `resources/sample_vaults/Tasks-Demo/_meta/templates/Test Data file.md`
- Add a Templater script mechanism to convert these markdown files to TypeScript
    - `resources/sample_vaults/Tasks-Demo/_meta/templater_scripts/convert_test_data_markdown_to_js.js`
        - The code to do the conversion
    - `resources/sample_vaults/Tasks-Demo/_meta/templates/convert_test_data_markdown_to_js.md`
        - Template file that executes the conversion
- Install the 'Broken Links' plugin to find and fix any accidentally broken links

**Note**: sometimes editing the Markdown can change the order of the fields in the generated `.ts` file - e.g. see f40fe328db094f26d4d8a09f75c4e78265ba04ad. This is irritating but harmless.

---

See the updated instructions in `Cache.test.ts` for how to use this new mechanism:

#### Test creation sequence

- Create a sample markdown file in Tasks demo vault (`root/Test Data/`) with the simplest content
to represent your test case. Choose a meaningful file name in snake case. See example in `Test Data/one_task.md`.
    - There is a Templater template that may help with creating a new file, for single-tasks cases:
      `resources/sample_vaults/Tasks-Demo/_meta/templates/Test Data file.md`
- Open any other note in Source or 'Live Preview' modes
- Run the command `Templater: Insert _meta/templates/convert_test_data_markdown_to_js.md`
    - Or type the short-cut `Ctrl + Cmd + Alt + T` / `Ctrl + Ctrl + Alt + T`
- This will convert all the files `root/Test Data/*.md` to test functions in `tests/Obsidian/__test_data__/*.ts`
- Run `yarn lint` to standardise the formatting in the generated TypeScript files.
- Use the data in the test with `readTasksFromSimulatedFile()`, the argument is the constant you
created in the previous step.
- Remember to commit the markdown file in the demo vault and the file with the simulated data.

---

## Motivation and Context

- Make creation of test data more streamlined and reliable




## How has this been tested?


- By running the new script and reviewing the generated files.





## Screenshots (if appropriate)

## Types of changes



Internal changes:


## Checklist




    -  _Comments in Cache.test.ts should be moved to the Contributing site at some point_

## Terms
